File: trie.py
class node:
    def __init__(self):
        self.term = 0
        self.to = [-1] * 26
 
class trie:
    def __init__(self):
        self.root = node()
    def add(self, s):
        p = [ord(i) - 97 for i in s]
        cur = self.root
        for i in p:
            if(cur.to[i] == -1):
                cur.to[i] = node()
            cur = cur.to[i]
        cur.term += 1
    def contain(self, s):
        p = [ord(i) - 97 for i in s]
        cur = self.root
        for i in p:
            if(cur.to[i] == -1):
                return False
            cur = cur.to[i]
        if(cur.term > 0):
            return True
        return False
    def delete(self, s):
        if(self.contain(s) == False):
            return False
        p = [ord(i) - 97 for i in s]
        cur = self.root
        for i in p:
            if(cur.to[i] == -1):
                return False
            cur = cur.to[i]
        cur.term -= 1
        return True

File: trie.cpp
#pragma GCC optimize("O3")
#pragma GCC target("avx2")

#include <iostream>
#include <vector>
#include <algorithm>
#include <string>
#include <cmath>
#include <set>
#include <cstring>
#include <deque>
#include <map>

#define iter vector<long long int>::iterator
#define input() ios::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL);

using namespace std;

typedef long long int ll;

ll inf = 1000000007;

struct node{
    ll term = 0;
    ll exist[26] = {0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0};
    node* to[26];
};

struct trie{
    node* root = new node();

    void add(string s){
        node* cur = root;
        for(auto i : s){
            ll c = i - 'a';
            if(cur->exist[c] == 0){
                cur->exist[c] = 1;
                cur->to[c] = new node();
            }
            cur = cur->to[c];
        }
        cur->term += 1;
    }
};

int main(){
    return 0;
}
